shap/maskers/_tabular.py

```diff
@@ -299,24 +308,47 @@ def __init__(self, data, max_samples=100, clustering="correlation"):
         super().__init__(data, max_samples=max_samples, clustering=clustering)
 
 
-class Impute(Masker): # we should inherit from Tabular once we add support for arbitrary masking
+class Impute(Tabular):
     """ This imputes the values of missing features using the values of the observed features.
 
     Unlike Independent, Gaussian imputes missing values based on correlations with observed data points.
     """
 
-    def __init__(self, data, method="linear"):
+    def __init__(self, data, max_samples=100, method="mean"):
         """ Build a Partition masker with the given background data and clustering.
 
         Parameters
         ----------
-        data : numpy.ndarray, pandas.DataFrame or {"mean: numpy.ndarray, "cov": numpy.ndarray} dictionary
+        data : numpy.ndarray, pandas.DataFrame
             The background dataset that is used for masking.
+
+        max_samples : int
+            The maximum number of samples to use from the passed background data. If data has more
+            than max_samples then shap.utils.sample is used to subsample the dataset. The number of
+            samples coming out of the masker (to be integrated over) matches the number of samples in
+            the background dataset. This means larger background dataset cause longer runtimes. Normally
+            about 1, 10, 100, or 1000 background samples are reasonable choices.
+
+        method : string or sklearn.impute object
+            If a string, then this is shorthand for the type of sklearn.impute object to generate.
+            Either a SimpleImputer or KNNImputer is used with default settings.
+            Mean, median, and mode refer to the supported SimpleImputer strategies.
+            For more finetuning, pass an already initialized sklearn.impute object.
+            Supported methods are:
+                mean - SimpleImputer with elements replaced by mean of feature in data.
+                median - SimpleImputer with elements replaced by median of feature in data.
+                mode - SimpleImputer with elements replaced by most frequent of feature in data.
+                knn - KNNImputer with elements replaced by mean value within 5 NNs of feature in data.
         """
-        if data is dict and "mean" in data:
-            self.mean = data.get("mean", None)
-            self.cov = data.get("cov", None)
-            data = np.expand_dims(data["mean"], 0)
+        methods = ["mean", "median", "mode", "knn"]
+        if isinstance(method, str):
+            if method not in methods:
+                raise NotImplementedError("Given imputation method is not supported.")
+            elif method == "knn":
+                impute = KNNImputer(missing_values=0)
+            else:
+                impute = SimpleImputer(missing_values=0, strategy=method)
+        else:
+            impute = method
```

Comment:
Seems to me that this should check whether `method` is an appropriate object (`sklearn.impute.SimpleImputer`, `sklearn.impute.KNNImputer`, etc.) rather than any other object that will result in an error when `self.impute.fit` is called later.